src/ucs/sys/sock.c

```diff
@@ -378,50 +381,51 @@ ucs_socket_handle_io_error(int fd, const char *name, ssize_t io_retval, int io_e
  *                         the array of iov.
  * @param [out] length_p   Pointer to the result length of user's data that was
  *                         sent/received.
- * @param [in]  io_retval  The result of the IO operation.
  * @param [in]  is_iov     Flag that specifies type of the operation (1 if vector
  *                         operation).
+ * @param [in]  io_retval  The result of the IO operation.
+ * @param [in]  io_errno   IO operation errno.
+ * @param [in]  err_cb     Error callback.
+ * @param [in]  err_cb_arg User's argument for the error callback.
  *
- * @return if the IO operation was successful - 1, otherwise - 0.
+ * @return if the IO operation was successful - UCS_OK, otherwise - error status.
  */
-static inline int
-ucs_socket_handle_io_success(int fd, const void *data, size_t count,
-                             size_t *length_p, int io_retval, int is_iov)
-{
-    if (ucs_likely(/* The IO operation is considered as successful if: */
-                   /* - the return value > 0 */
-                   (io_retval > 0) ||
-                   /* - the return value == 0 and the user's data length == 0
-                    *   (the number of the iov array buffers == 0 or the total
-                    *   length of the iov array buffers == 0) */
-                   ((io_retval == 0) &&
-                    (!is_iov ? (count == 0) :
-                     ((count == 0) ||
-                      (ucs_iov_total_length((const struct iovec*)data,
-                                            count) == 0)))))) {
+static inline ucs_status_t
+ucs_socket_handle_io(int fd, const void *data, size_t count,
+                     size_t *length_p, int is_iov, int io_retval,
+                     int io_errno, const char *name,
+                     ucs_socket_io_err_cb_t err_cb, void *err_cb_arg)
+{
+    /* The IO operation is considered as successful if: */
+    if (ucs_likely(io_retval > 0)) {
+        /* - the return value > 0 */
         *length_p = io_retval;
-        return 1;
+        return UCS_OK;
+    } else if (io_retval == 0) {
```

Comment:
this looks better
can remove else since there is already return

Reply:
sorry, I'm not sure that I follow you suggestion
which `if` statement should be removed?

Reply:
remove `else`

Reply:
done